File: src/prod/src/data/txnreplicator/logrecordlib/CopyStage.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace ktl;

namespace Data
{
    namespace LogRecordLib
    {
        namespace CopyStage
        {
            void WriteToTextWriter(Common::TextWriter & w, Enum const & val)
            {
                switch (val)
                {
                case Invalid:
                    w << L"Invalid"; return;
                case CopyMetadata:
                    w << L"CopyMetadata"; return;
                case CopyNone:
                    w << L"CopyNone"; return;
                case CopyState:
                    w << L"CopyState"; return;
                case CopyProgressVector:
                    w << L"CopyProgressVector"; return;
                case CopyFalseProgress:
                    w << L"CopyFalseProgress"; return;
                case CopyScanToStartingLsn:
                    w << L"CopyScanToStartingLsn"; return;
                case CopyLog:
                    w << L"CopyLog"; return;
                case CopyDone:
                    w << L"CopyDone"; return;
                default:
                    Common::Assert::CodingError("Unknown CopyStage:{0} in Replicator", val);
                }
            }

            ENUM_STRUCTURED_TRACE(CopyStage, Invalid, LastValidEnum)
        }
    }
}

File: src/prod/src/managed/Microsoft.ServiceFabric.Data.Impl/Replicator/CopyStage.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace Microsoft.ServiceFabric.Replicator
{
    /// <summary>
    /// The copy stage.
    /// </summary>
    internal enum CopyStage
    {
        /// <summary>
        /// The invalid.
        /// </summary>
        Invalid = 0, 

        /// <summary>
        /// The copy metadata.
        /// </summary>
        CopyMetadata = 1, 

        /// <summary>
        /// The copy none.
        /// </summary>
        CopyNone = 2, 

        /// <summary>
        /// The copy state.
        /// </summary>
        CopyState = 3, 

        /// <summary>
        /// The copy progress vector.
        /// </summary>
        CopyProgressVector = 4, 

        /// <summary>
        /// The copy false progress.
        /// </summary>
        CopyFalseProgress = 5, 

        /// <summary>
        /// The copy scan to starting lsn.
        /// </summary>
        CopyScanToStartingLsn = 6, 

        /// <summary>
        /// The copy log.
        /// </summary>
        CopyLog = 7, 

        /// <summary>
        /// The copy done.
        /// </summary>
        CopyDone = 8, 
    }
}